File: boat-sample/src/test/java/sample/xyz/srclab/common/bean/BeanSample.java
package sample.xyz.srclab.common.bean;

import org.testng.annotations.Test;
import xyz.srclab.common.bean.Beans;
import xyz.srclab.common.test.TestLogger;

public class BeanSample {

    private static final TestLogger logger = TestLogger.DEFAULT;

    @Test
    public void testBean() {
        A a = new A();
        a.setP1("1");
        a.setP2("2");
        B b = Beans.copyProperties(a, new B());
        int b1 = b.getP1();
        int b2 = b.getP2();
        //1
        logger.log("b1: {}", b1);
        //2
        logger.log("b1: {}", b2);
    }

    public static class A {
        private String p1;
        private String p2;

        public String getP1() {
            return p1;
        }

        public void setP1(String p1) {
            this.p1 = p1;
        }

        public String getP2() {
            return p2;
        }

        public void setP2(String p2) {
            this.p2 = p2;
        }
    }

    public static class B {
        private int p1;
        private int p2;

        public int getP1() {
            return p1;
        }

        public void setP1(int p1) {
            this.p1 = p1;
        }

        public int getP2() {
            return p2;
        }

        public void setP2(int p2) {
            this.p2 = p2;
        }
    }
}

File: boat-sample/src/test/kotlin/sample/kotlin/xyz/srclab/core/bean/BeanSample.kt
package sample.kotlin.xyz.srclab.core.bean

import org.testng.annotations.Test
import xyz.srclab.common.bean.copyProperties

class BeanSample {

    @Test
    fun testBean() {
        val a = A()
        a.p1 = "1"
        a.p2 = "2"
        val b = a.copyProperties(B())
        val b1 = b.p1
        val b2 = b.p2
        //1
        logger.log("b1: {}", b1)
        //2
        logger.log("b1: {}", b2)
    }

    class A {
        var p1: String? = null
        var p2: String? = null
    }

    class B {
        var p1 = 0
        var p2 = 0
    }

    companion object {
        private val logger = TestLogger.DEFAULT
    }
}